Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6452,6 +6452,8 @@ void WebPageProxy::didFailProvisionalLoadForFrameShared(Ref<WebProcessProxy>&& p
 
     ASSERT(!m_failingProvisionalLoadURL);
     m_failingProvisionalLoadURL = provisionalURL;
+    URL failedURL { provisionalURL };
+    bool canFallbackToHTTP = frame.isMainFrame() && error.errorRecoveryMethod() == ResourceError::ErrorRecoveryMethod::HTTPFallback && failedURL.protocolIs("https"_s);
```

Comment:
Can we move the scope of this into the WebsitePolicies if statement below and add a nested if statement?  That'll make the code do less in the common case and be better organized so it'll be more clear what these things are used for.

Reply:
Sure. I moved it here so it is shared between both blocks below (HTTPSOnly and HTTPSFirst), but I can also just duplicate this within each block.